Objects/complexobject.c

```diff
@@ -53,11 +53,66 @@ _Py_c_neg(Py_complex a)
 }
 
 Py_complex
-_Py_c_prod(Py_complex a, Py_complex b)
+_Py_c_prod(Py_complex z, Py_complex w)
 {
     Py_complex r;
-    r.real = a.real*b.real - a.imag*b.imag;
-    r.imag = a.real*b.imag + a.imag*b.real;
+    double a = z.real, b = z.imag, c = w.real, d = w.imag;
+    double ac = a*c, bd = b*d, ad = a*d, bc = b*c;
+
```

Comment:
Maybe put the `Py_complex` just above `r.real`? (or do we follow strict ANSI C?)

Reply:
Not sure I get this point, could you be more precise?  I'm trying to follow PEP 7 here.

Reply:
In ANSI C, you need to put the declarations at the top of the block the variable is being used. PEP 7 tells me that we use C11 for Python 3.11 and later and AFAIR, we don't have this constraint anymore. However, I'm wondering whether this code counts as an "old" code or not and which convention it should follow.

Reply:
To be clear, I suggested doing:

```C
{
    double a = .....
    double ac = ....

    Py_complex r;
    r.real = ...
    r.imag = ...
	...
}
```

Reply:
How this looks:
```diff
diff --git a/Objects/complexobject.c b/Objects/complexobject.c
index c5f40b0965..e201cd06c3 100644
--- a/Objects/complexobject.c
+++ b/Objects/complexobject.c
@@ -55,12 +55,9 @@ _Py_c_neg(Py_complex a)
 Py_complex
 _Py_c_prod(Py_complex z, Py_complex w)
 {
-    Py_complex r;
     double a = z.real, b = z.imag, c = w.real, d = w.imag;
     double ac = a*c, bd = b*d, ad = a*d, bc = b*c;
-
-    r.real = ac - bd;
-    r.imag = ad + bc;
+    Py_complex r = {ac - bd, ad + bc};
 
     /* Recover infinities that computed as nan+nanj.  See e.g. the C11,
        Annex G.5.2, routine _Cmultd(). */
```
?